app/controllers/consent_responses_controller.rb

```diff
@@ -13,22 +13,61 @@ def create
       parent_phone: "07412 345678",
       parent_email: "test.parent@example.com",
       parent_relationship: "mother",
-      consent: "given",
       route: "website",
       health_questions: ConsentResponse::HEALTH_QUESTIONS
         .fetch(:hpv)
         .map { |question| { question: } }
     )
 
-    redirect_to action: :edit_questions
+    redirect_to action: :edit_agree
   end
 
   def update
-    @draft_consent_response.health_questions.each_with_index do |hq, index|
-      hq.merge! consent_response_health_questions_params["question_#{index}"]
+    if consent_response_agree_params.present?
+      @draft_consent_response.assign_attributes(consent_response_agree_params)
+      if @draft_consent_response.save(context: :edit_consent)
+        # Reset the reason for refusal so the user has to pick it again.
+        # Otherwise it will be pre-filled with the previous value.
+        @draft_consent_response.update! reason_for_refusal: nil
+
+        if @draft_consent_response.consent_given?
+          redirect_to action: :edit_questions
+        elsif @draft_consent_response.consent_refused?
+          redirect_to action: :edit_reason
+        else
+          redirect_to action: :edit_confirm
+        end
+      else
+        render :edit_agree
+      end
+    end
+
+    if consent_response_reason_params.present?
```

Comment:
My expectation here, knowing that we have multi-page editing, is that this method would choose one of the pages to respond to, either "Do they agree ...", the health questions or the reason for refusal, but because it's not using `elsif` I'm suddenly not sure ... is that intentional? Could updating these params overlap, and is that why that's done.

I can see cases where we might do that ... perhaps to facilitate an API call or if we have a form that updates more things at once ... but when mixed in with these renders and redirects I think it's a bit confusing. Also doesn't help that this method is getting long.

This is tricky ... happy to roll-forward with this and address it later.

Reply:
Yeah, it should probably be elsif, or we should break this up into separate methods.